### ISSUE TITLE
GeM pooling parameter

### ISSUE THREAD

**Issue by andrefaraujo** [almazan/deep-image-retrieval#9]
Hi @almazan ,

For your experiment trained on the Google Landmarks dataset 2018 (codenamed `Resnet101-AP-GeM-LM18`): could you share to which value the GeM pooling `p` parameter converged to?

If you could additionally share learning curve showing the evolution of p over the training run, that would be even better :)

Thanks!

**Comment by almazan** [comment]
Hi @andrefaraujo,

For that experiment, the GeM pooling `p` parameter converged to `2.99`.

Unfortunately, I don't have its evolution because I don't have access to the intermediate snapshots anymore. However, given that we used `3` as the starting value, my guess is that that curve is pretty flat :D

Interestingly, all models converged to a similar value, not really moving away from the starting point.

I hope it helped. 

Cheers!

**Comment by andrefaraujo** [comment]
Very interesting, thanks!

**Comment by 15122901359** [comment]
Hello, can you share the multistaged_backpropagation training process？How do you calculate the derivative of the similarity matrix S and matrix D?I calculated it automatically through pytorch, but the parameter update seems to be a bit problematic.
`            desc_db = Variable(torch.cuda.FloatTensor(desc_db),requires_grad=True)
            scores = torch.matmul(desc_db,desc_db.t())
            vaild_index = np.arange(batch_size*index,batch_size*(index+1),1)
            Y = np.array(Y_all)[vaild_index][:,vaild_index]
            Y = torch.cuda.FloatTensor(np.array(Y))
            rank_loss = criterion(scores, Y)
            # rank_loss = Variable(rank_loss, requires_grad=True)
            rank_loss.backward()
            loss += rank_loss.item()
            net.train()
            for i,img in enumerate(imgs):
                img = Variable(img.cuda(),requires_grad=True)
                desc = net(img.unsqueeze(dim=0))
                one_grad = desc_db.grad[i].unsqueeze(0)
                desc.unsqueeze(0).backward(one_grad)
            optimizer.step()
            scheduler_mul.step()
            optimizer.zero_grad()
            lr = scheduler_mul.get_lr()[0]`
